### PR TITLE
fix middleware to work with array upload

### PR DESCRIPTION
**Pull request by ctf0**

### CODE REVIEW

File: src/ImageOptimizerServiceProvider.php [hunk 1]
```diff
@@ -25,5 +25,7 @@ public function boot()
     public function register()
     {
         $this->mergeConfigFrom(__DIR__.'/../config/image-optimizer.php', 'image-optimizer');
+
+        $this->app['router']->aliasMiddleware('optimizeImages', \Spatie\LaravelImageOptimizer\Middlewares\OptimizeImages::class);
```
Comment (freekmurze) [inline src/ImageOptimizerServiceProvider.php]: This is a kinda of breaking. Current users of this package which did opt to not use the middleware will suddenly have the middleware installed.

Could you remove this change and restore the readme to it's current state?

File: src/Middlewares/OptimizeImages.php [hunk 2]
```diff
@@ -12,8 +12,14 @@ public function handle($request, Closure $next)
...
         $optimizerChain = app(OptimizerChain::class);
 
-        collect($request->allFiles())->each(function (UploadedFile $file) use ($optimizerChain) {
-            $optimizerChain->optimize($file->getPathname());
+        collect($request->allFiles())->each(function ($file) use ($optimizerChain) {
+            if (is_array($file)) {
+                collect($file)->each(function (UploadedFile $one) use ($optimizerChain) {
+                    $optimizerChain->optimize($one->getPathname());
+                });
+            } else {
```
Comment (freekmurze) [inline src/Middlewares/OptimizeImages.php]: Could you rewrite this so an `else` isn't needed?

Comment (ctf0) [reply src/Middlewares/OptimizeImages.php]: > it will give error cuz it will run after the loop has finished.

Comment (freekmurze) [reply src/Middlewares/OptimizeImages.php]: If file isn't an array, make it an array so you'll only need one path for processing it.

Comment (ctf0) [reply src/Middlewares/OptimizeImages.php]: updated, however i found a strange behavior with `$request->allFiles()`,

- if you have a form with single file upload ex.`avatar`
- using `$request->allFiles()` will return empty array

File: src/Middlewares/OptimizeImages.php [hunk 3]
```diff
@@ -12,7 +12,7 @@ public function handle($request, Closure $next)
     {
         $optimizerChain = app(OptimizerChain::class);
 
-        collect($request->allFiles())->each(function (UploadedFile $file) use ($optimizerChain) {
+        collect(array_flatten($request->allFiles()))->each(function (UploadedFile $file) use ($optimizerChain) {
```
Comment (freekmurze) [inline src/Middlewares/OptimizeImages.php]: It's more readable to just use the `flatten` collection method here.

**Comment by dungnh** [comment]
@freekmurze I'm sorry if it bothers you, but I want to ask when this will be merged.

**Comment by freekmurze** [comment]
Thanks!

**Pull request merged**